components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/api/APIDefinition.java

```diff
@@ -55,14 +55,13 @@
     Map<String, Scope> getScopes(String resourceConfigsJSON) throws APIManagementException;
 
     /**
-     * Get Scopes Extracts from the API definition
+     * Get Scopes Extracted from the API definition
      *
      * @param resourceConfigJSON resource json
      * @return Map of the scopes
      * @throws APIManagementException If error occurs while parsing swagger resources.
      */
-    //todo: keep only single getScopes method after .yamls in same format
-    Map<String, String> getScope(String resourceConfigJSON) throws APIManagementException;
+    Map<String, Scope> getScope(String resourceConfigJSON) throws APIManagementException;
```

Comment:
What is the different between this method and getScopes? If this returns a scope, we don't need a map here.

Reply:
this is to refactor in later step. Currently we are using getScopes to retrieve scopes when vendor extensions are present. This getScope method is to retrieve scopes when OAuth2Security definitions are present. We might need to remove getSopes methods. I'l refactor this anyway.